### PR TITLE
[W5][W14-1]Lee Tze Ting

### PR DESCRIPTION
**Pull request by halcyoneee**
Added a new `deletemulti` command which deletes a list of people that the user specify. It deletes in descending order.

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -111,6 +111,26 @@ Deletes the 2nd person in the address book.
 `delete 1` +
 Deletes the 1st person in the results of the `find` command.
 
+== Deleting multiple persons : `deletemulti`
+
+Deletes the specified list of persons from the address book. Irreversible. +
+Format: `deletemulti INDEX`
```
Comment (eugenefdw) [inline docs/UserGuide.adoc]: Could change to deletemulti INDEX [INDEX...]
so it's more clear that it accepts multi param in the user guide

File: src/seedu/addressbook/commands/DeleteMultiCommand.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.util.ArrayList;
+
+public class DeleteMultiCommand extends Command {
+
+    public static final String COMMAND_WORD = "deletemulti";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes multiple persons identified by the index number used in the last person listing.\n"
+            + "Parameters: INDEX\n"
```
Comment (eugenefdw) [inline src/seedu/addressbook/commands/DeleteMultiCommand.java]: Ditto from above, not obvious that its multi-param by the wording

Comment (eugenefdw) [reply src/seedu/addressbook/commands/DeleteMultiCommand.java]: INDEX [INDEX...]

File: docs/UserGuide.adoc (hunk 1 above)
Comment (halcyoneee) [reply docs/UserGuide.adoc]: Thanks for suggesting that, was wondering how to show that.

File: src/seedu/addressbook/commands/DeleteMultiCommand.java (hunk 2 above)
Comment (halcyoneee) [reply src/seedu/addressbook/commands/DeleteMultiCommand.java]: Same as above, thanks for the suggestion.

**Comment by lallanachang** [comment]
Well done! Seems to be a good command that can be added to the final project.
I just have one small question about the coding style for example indentation.
Such as the method between lines 67-72 in DeleteMultiCommandTest.java, it just looks a bit inconsistent for me..

**Comment by halcyoneee** [comment]
> Well done! Seems to be a good command that can be added to the final project.
> I just have one small question about the coding style for example indentation.
> Such as the method between lines 67-72 in DeleteMultiCommandTest.java, it just looks a bit inconsistent for me..

@lallanachang Though it seems inconsistent with other methods but it is consistent with similar codes at line 37 to 45. If you want to be really consistent and you should, then you should ensure it word wrap at line 80~. As the coding style now isn't that obvious when you should do a new line.

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -228,6 +259,25 @@ private int parseArgsAsDisplayedIndex(String args) throws ParseException, Number
...
+     *
+     * @param args arguments string to parse as a list of index numbers
+     * @return the parsed list of index numbers
+     * @throws ParseException if no region of the args string could be found for the index
+     * @throws NumberFormatException the args string region is not a valid number
+     */
+    private String[] parseArgsAsDisplayedIndices(String args) throws ParseException, NumberFormatException {
+        final String[] extractedIndices = args.trim().split(" ");
+        for (String index: extractedIndices) {
+            if (!index.matches("\\d+")){
```
Comment (kylase-learning) [inline src/seedu/addressbook/parser/Parser.java:273]: Regular expression is not cheap to compute, one way to check whether it is a number is to cast a string to number and if it raises an exception, it is not a number.

**Review by kylase-learning — COMMENTED** [review]
Well done on implementing the JUnit test for your new command 👍.

Great to see you have submitted your changes via your own branch and done the documentation.

You might want to try to write the test first before implementing the functionality to adopt TDD workflow.

You have demonstrated the progress of this exercise through the commits you have done. Maybe, as you get more comfortable, your commits can be done in batch, by committing the test, the implementation and the relevant documentation together, instead of committing changes in each file separately. This way, it is easier to navigate through commits in events you need to revert to certain state of the code, and all its related dependencies (such as documentation, tests) are in sync. Otherwise, for example, you can revert the commit for the code change but not the documentation (if the documentation is written prior to the code).

Please close this PR after reading comments. Thanks.

**Pull request closed without merging**